### ISSUE TITLE
Chore: make serialization of backend objects in the frontend systematic

### ISSUE THREAD

**Issue by raphj** [tracim/tracim#3742]
## Description and expectations

It has been decided that objects from the backend should be serialized in the frontend: snake_case should be converted to camelCase (here, serialization does not mean: turned into a string or another format that can be used for transfer or storage; it's more a conversion).

This serialization makes it less easy to grep the frontend code for API usage or properties, but this is common practice to ease consistency with the rest of the code.

However, the current situation has several issues:

 - this is not done everywhere
 - some functions were made to handle both versions
 - it is not always clear if the object we are going to handle is serialized or not
 - some fields change name: `workspace_id` or `user_id` become `id` (but not always, `userId` also exists)
 - serializers are more or less manually written, and serialized fields do not always match the original field. For instance, in ReduxTLMDispatcher, field author of the notifications is the author's public_name, instead of the full user object returned by the  backend
 - serializers need to be manually updated to handle new fields, and one can be surprised to not find new fields if they forget to update the serializers.
 - there are serializer for each kind of objects.

I see two ways to tackle the issue:
1. by keeping the serialization
  - we should probably serialize at the interface only. Serialization should never happen at a deeper level
  - to avoid the need to refactor the whole frontend, serialization could keep the original fields, so produced objects could be used by both the current code and new code
 -  Maybe we should have a generic serializer that deeply renames all snake_case fields to camelCase for every objects.
2. By not serializing anymore (this is probably harder since we already have code that assume serialized objects).

**Comment by grignards** [comment]
Apart from enforcing our coding style for API objects the serialization could also be used to validate the input we get from the backend (as the backend does for what we give to it).
To do both (validation + serialization) we could use a validation library, for example [Yup](https://github.com/jquense/yup). To avoid writing all schemas manually we could use a tool like https://github.com/kristianmandrup/schema-to-yup to generate yup schemas from the HTTP API documentation.

**Comment by lebouquetin** [comment]
I'm afraid to make automatic mapping from backend API to frontend, because it means we'll loose all our knowledge about REST API. It will also make chagnes more transparent for tracim developers and I fear we break APIs without to identify that we broke it (common problem of development when things are not static)

**Comment by grignards** [comment]
> I'm afraid to make automatic mapping from backend API to frontend, because it means we would loose our knowledge of our REST API. It will also make chagnes more transparent for tracim developers and I fear we break APIs without to identify that we broke it (common problem of development when things are not static)
> 
> (edit) : the idea is interesting but the REST API is a user interface and must be managed as a final product

Yes, when I talked about generating the schemas automatically I did not necessarily envisioned to always do it (apart of the issue of unknown API break we also have the issue that frontend development is frequently started before the backend APIs are available). But this would help at the starting phase to avoid writing several dozens of schemas manually.

**Comment by raphj** [comment]
Another instance of this issue: #3578 is partly caused by the fact that a caller sets the `type` property on an object representing contents while the callee expects to have `content_type`. We are being bitten by the lack of type checking and inconsistencies in how we represent things in the frontend.
